router/article.js

```diff
@@ -0,0 +1,109 @@
+import express from "express";
+import { PrismaClient } from "@prisma/client";
+import { BadRequestError, InternalServerError } from "../middleware/error.js";
+
+const prisma = new PrismaClient();
+const router = express.Router();
+
+// 유효성 검사
+import { validateArticle } from "../middleware/validate.js";
+
+// 게시글 등록 + 목록 조회
+router
+  .route("/")
+  .post(validateArticle, async (req, res) => {
+    try {
+      const { title, content } = req.body;
+      const article = await prisma.article.create({
+        data: { title, content },
+      });
+      res.status(201).json(article);
+    } catch (e) {
+      next(new InternalServerError("게시글 등록에 실패했습니다."));
+    }
+  })
+  .get(async (req, res) => {
+    try {
+      const offset = parseInt(req.query.offset) || 0;
+      const limit = parseInt(req.query.limit) || 10;
+      const search = req.query.search?.toString() || "";
+
+      const article = await prisma.article.findMany({
+        where: {
+          OR: [
+            { title: { contains: search } },
+            { content: { contains: search } },
+          ],
+        },
+        skip: offset,
+        take: limit,
+        orderBy: {
+          createdAt: "desc",
+        },
+        select: {
+          id: true,
+          title: true,
+          content: true,
+          createdAt: true,
+        },
+      });
+      if (!article) next(new BadRequestError("게시글을 찾을 수 없습니다."));
```

Comment:
빈 배열도 truty값이기 때문에 Not Null Assertion으로 에상하신 동작이 나타나지 않을거에요.
만약 의도하신 게 찾을 수 없다는 에러라면 404를 던져주세요!

```suggestion
      if (article.length === 0) next(new NotFoundError("게시글을 찾을 수 없습니다."));
```

Reply:
그리고 보통 리스트 조회의 응답에는 빈 배열이라면 404대신 200에 `[]` 빈 배열을 반환하곤 합니다.